libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -231,6 +232,45 @@ bool StaticAnalyzer::visit(InlineAssembly const& _inlineAssembly)
 	return true;
 }
 
+bool StaticAnalyzer::visit(BinaryOperation const& _operation)
+{
+	if (
+		_operation.rightExpression().annotation().isConstant &&
+		(_operation.getOperator() == Token::Div || _operation.getOperator() == Token::Mod)
+	)
+		if (auto result = dynamic_pointer_cast<RationalNumberType const>(
+			ConstantEvaluator(m_errorReporter).evaluate(_operation.rightExpression())
+		))
+			if (result->literalValue(nullptr) == u256(0))
+				m_errorReporter.typeError(
+					_operation.location(),
+					"Division by zero."
+				);
+
+	return true;
+}
+
+bool StaticAnalyzer::visit(FunctionCall const& _functionCall)
+{
+	if (_functionCall.annotation().kind == FunctionCallKind::FunctionCall)
+	{
+		auto functionType = dynamic_pointer_cast<FunctionType const>(_functionCall.expression().annotation().type);
+		if (functionType->kind() == FunctionType::Kind::AddMod || functionType->kind() == FunctionType::Kind::MulMod)
+		{
+			solAssert(_functionCall.arguments().size() > 2, "");
+			if (auto result = dynamic_pointer_cast<RationalNumberType const>(
+				ConstantEvaluator(m_errorReporter).evaluate(*(_functionCall.arguments())[2])
+			))
+				if (result->literalValue(nullptr) == u256(0))
```

Comment:
Will this not categorise `addmod(0, 0, 1)` and `mulmod(0, 0, 1)` as failure ?

Reply:
Same as above - I'll change the cpp variable name to clarify that only the third argument is evaluated and checked.